artcommon/artcommonlib/util.py

```diff
@@ -85,6 +88,58 @@ def merge_objects(a, b):
     return c
 
 
+def is_future_release_date(date_str):
+    """
+    If the input date is in future then return True elase False
+    """
+    try:
+        target_date = datetime.strptime(date_str, "%Y-%m-%d")
+    except ValueError:
+        return False
+    current_date = datetime.now()
+    if target_date > current_date:
+        return True
+    else:
+        return False
+
+
+def get_assembly_release_date(assembly, group):
+    """
+    Get assembly release release date from release schedule API
+    """
+    release_schedules = requests.get(f'{RELEASE_SCHEDULES}/{group}.z/?fields=all_ga_tasks', headers={'Accept': 'application/json'})
+    for release in release_schedules.json()['all_ga_tasks']:
+        if assembly in release['name']:
+            assembly_release_data = datetime.strptime(release['date_start'], "%Y-%m-%d").strftime("%Y-%b-%d")
+            break
+    return assembly_release_data
```

Comment:
Let's also factor the case when assembly is not found

Reply:
add define with None